### PR TITLE
refs #CHRONAM-413 - add chronam and lccn tags for cloudflare

### PR DESCRIPTION
**Pull request by johnscancella**

### CODE REVIEW

File: core/utils/utils.py [hunk 1]
```diff
@@ -276,3 +276,14 @@ def validate_bib_dir():
         return settings.BIB_STORAGE
     else:
         return None
+
+def cache_tag(response, tag_name):
+    """
+    Adds Cache-Tag header or appends to it if it already exists. This is
+    useful for cloudflare so that we can purge based on the tag
+    """
+    if response.has_header('Cache-Tag'):
```
Comment (acdha) [inline core/utils/utils.py]: Think it's worth documenting that the caller is expected to escape values?

https://support.cloudflare.com/hc/en-us/articles/206596608-How-to-Purge-Cache-Using-Cache-Tags-Enterprise-only-

File: core/views/browse.py [hunk 2]
```diff
@@ -53,8 +53,9 @@ class SelectYearForm(django_forms.Form):
     page_title = "Browse Issues: %s" % title.display_name
     page_name = "issues"
     crumbs = create_crumbs(title)
-    return render_to_response('issues.html', dictionary=locals(),
+    response = render_to_response('issues.html', dictionary=locals(),
                               context_instance=RequestContext(request))
+    return cache_tag(response, "lccn=%s" % lccn)
```
Comment (acdha) [inline core/views/browse.py]: This usage feels like it should be a decorator to match `cache_page`

Comment (acdha) [reply core/views/browse.py]: Also, the trailing indentation should match the parentheses following PEP-8

File: core/utils/utils.py [hunk 3]
```diff
@@ -276,3 +276,14 @@ def validate_bib_dir():
         return settings.BIB_STORAGE
     else:
         return None
+
+def cache_tag(response, tag_name):
```
Comment (acdha) [inline core/utils/utils.py]: What do you think about naming this function `add_cache_tag`?

File: core/views/browse.py (hunk 2 above)
Comment (johnscancella) [reply core/views/browse.py]: How would the decorator get the LCCN?

File: core/utils/utils.py (hunk 3 above)
Comment (johnscancella) [reply core/utils/utils.py]: 👍

File: core/utils/utils.py (hunk 1 above)
Comment (johnscancella) [reply core/utils/utils.py]: 👍

**Pull request merged**